pkg/pbflow/proto.go

```diff
@@ -99,13 +101,32 @@ func FlowToPB(fr *model.Record, s *ovnobserv.SampleDecoder) *Record {
 		pbflowRecord.Network.DstAddr = &IP{IpFamily: &IP_Ipv4{Ipv4: model.IntEncodeV4(fr.Id.DstIp)}}
 	}
 	if s != nil {
-		seen := make(map[string]bool)
+		seen := make(map[ovnmodel.NetworkEvent]bool)
+		pbflowRecord.NetworkEventsMetadata = make([]*NetworkEvent, 0)
 		for _, metadata := range fr.Metrics.NetworkEvents {
 			if !model.AllZerosMetaData(metadata) {
 				if md, err := s.DecodeCookie8Bytes(metadata); err == nil {
 					protoLog.Debugf("Network Events Metadata %v decoded Cookie: %v", metadata, md)
 					if !seen[md] {
-						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, md)
+						e := ovnmodel.ACLEvent{NetworkEvent: md}
+						pbEvents := []*NetworkEvent{
+							{
+								Events: map[string]string{"Action": e.Action},
+							},
+							{
+								Events: map[string]string{"Feature": e.Actor},
+							},
+							{
+								Events: map[string]string{"Name": e.Name},
+							},
+							{
+								Events: map[string]string{"Namespace": e.Namespace},
+							},
+							{
+								Events: map[string]string{"Direction": e.Direction},
+							},
+						}
+						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, pbEvents...)
```

Comment:
This shouldn't be one map per single event field, but one map per event (with all event fields in the same map)

```suggestion
						e := ovnmodel.ACLEvent{NetworkEvent: md}
						pbEvent := NetworkEvent{
								Events: map[string]string{
								    "Action": e.Action,
								    "Feature": e.Actor,
								    "Name": e.Name,
							            "Namespace": e.Namespace,
								    "Direction": e.Direction,
							},
						}
						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, pbEvents)
```

right?

Reply:
yes I addressed this change